services/motion/builtin/builtin_test.go

```diff
@@ -698,811 +700,252 @@ func TestMoveOnMapTimeout(t *testing.T) {
-		t.Parallel()
-		injectedMovementSensor, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, nil, 5)
-		defer ms.Close(ctx)
-		success, err := ms.MoveOnGlobe(
-			ctx,
-			fakeBase.Name(),
-			geo.NewPoint(math.NaN(), 0),
-			0,
-			injectedMovementSensor.Name(),
-			nil,
-			nil,
-			nil,
-		)
-		test.That(t, err, test.ShouldBeError, errors.New("destination may not contain NaN"))
-		test.That(t, success, test.ShouldBeFalse)
+		req := motion.MoveOnGlobeReq{
+			ComponentName:      kb.Name(),
+			Destination:        dst,
+			MovementSensorName: injectedMovementSensor.Name(),
+			MotionCfg:          motionCfg,
+			Extra:              tc.extra,
+		}
+		executionID, err := ms.MoveOnGlobe(ctx, req)
+		test.That(t, err, test.ShouldBeNil)
 
-		success, err = ms.MoveOnGlobe(
-			ctx,
-			fakeBase.Name(),
-			geo.NewPoint(0, math.NaN()),
-			0,
-			injectedMovementSensor.Name(),
-			nil,
-			nil,
-			nil,
-		)
-		test.That(t, err, test.ShouldBeError, errors.New("destination may not contain NaN"))
-		test.That(t, success, test.ShouldBeFalse)
+		timeoutCtx, timeoutFn := context.WithTimeout(ctx, time.Minute*5)
+		defer timeoutFn()
+		err = pollUntilMOGSuccessOrError(timeoutCtx, ms, motion.PlanHistoryReq{
+			ComponentName: req.ComponentName,
+			ExecutionID:   executionID,
+			LastPlanOnly:  true,
+		})
 
-		success, err = ms.MoveOnGlobe(
-			ctx,
-			fakeBase.Name(),
-			geo.NewPoint(math.NaN(), math.NaN()),
-			0,
-			injectedMovementSensor.Name(),
-			nil,
-			nil,
-			nil,
-		)
-		test.That(t, err, test.ShouldBeError, errors.New("destination may not contain NaN"))
-		test.That(t, success, test.ShouldBeFalse)
-	})
+		if tc.expectedSuccess {
+			test.That(t, err, test.ShouldBeNil)
+		} else {
+			test.That(t, err.Error(), test.ShouldEqual, tc.expectedErr)
+		}
+	}
 
-	t.Run("is able to reach a nearby geo point with empty values", func(t *testing.T) {
-		injectedMovementSensor, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, nil, 5)
-		defer ms.Close(ctx)
-		success, err := ms.MoveOnGlobe(
-			ctx,
-			fakeBase.Name(),
-			dst,
-			0,
-			injectedMovementSensor.Name(),
-			nil,
-			nil,
-			extraSmooth,
-		)
-		test.That(t, err, test.ShouldBeNil)
-		test.That(t, success, test.ShouldBeTrue)
-	})
+	for _, tc := range testCases {
+		c := tc // needed to workaround loop variable not being captured by func literals
+		t.Run(c.name, func(t *testing.T) {
+			t.Parallel()
+			testFn(t, c)
+		})
+	}
+}
 
-	t.Run("is able to reach a nearby geo point with a requested NaN heading", func(t *testing.T) {
-		injectedMovementSensor, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, nil, 5)
-		defer ms.Close(ctx)
-		success, err := ms.MoveOnGlobe(
-			ctx,
-			fakeBase.Name(),
-			dst,
-			math.NaN(),
-			injectedMovementSensor.Name(),
-			nil,
-			nil,
-			extraSmooth,
-		)
-		test.That(t, err, test.ShouldBeNil)
-		test.That(t, success, test.ShouldBeTrue)
-	})
+func TestObstacleDetection(t *testing.T) {
```

Comment:
can we call this TestObstacleReplanning and the above func TestPositionalReplanning?

Reply:
done